src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,112 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private Object[] array = new Object[DEFAULT_CAPACITY];
+
     @Override
     public void add(T value) {
-
+        for (int i = 0; i < array.length; i++) {
+            if (size < array.length) {
+                array[size] = value;
+                size++;
+                return;
+            } else {
+                expandArray();
+                array[size + 1] = value;
+            }
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        if (array.length < size + 1) {
+            expandArray();
+        }
+        Object[] newArray = new Object[array.length];
+        System.arraycopy(array, index, newArray, index + 1, size - index);
```

Comment:
тут можна одним arraycopy зробити 
```suggestion
      System.arraycopy(values, index, values, index + 1, size - index);
        values[index] = value;
        size++;
```